src/values/AbstractObjectValue.js

```diff
@@ -768,16 +768,11 @@ export default class AbstractObjectValue extends AbstractValue {
             let generator = this.$Realm.generator;
             invariant(generator);
 
-            if (typeof P === "string") {
-              generator.emitStatement(
-                [Receiver, new StringValue(this.$Realm, P), V],
-                createOperationDescriptor("ABSTRACT_OBJECT_SET_PARTIAL")
-              );
-            } else {
+            if (typeof P !== "string" && !(P instanceof StringValue)) {
               // Coercion can only have effects on anything reachable from the key.
               Havoc.value(this.$Realm, P);
-              generator.emitStatement([Receiver, P, V], createOperationDescriptor("ABSTRACT_OBJECT_SET_PARTIAL"));
```

Comment:
I think you may have removed the only references to ABSTRACT_OBJECT_SET_PARTIAL. If so, please remove its definition as well.